### PR TITLE
Move address book to profile

### PR DESCRIPTION
**Pull request by bfollington**
Resolves https://github.com/subconsciousnetwork/subconscious/issues/478
Resolves https://github.com/subconsciousnetwork/subconscious/issues/555

# Changes

-  [x] Address book view removed
-  [x] "Add User" button moved to your profile in the top-right corner
    - [x] Ensure only visible on YOUR profile
- [x] Sync after following a user
- [x] Unfollowing is possible from the list on your profile for quick management
    - [x] Ensure only visible on YOUR profile
- [x] Sort following list by petname (just needed _an_ ordering, not precious on what it is)
- [x] Find a new home for your QR code

# Demo

https://user-images.githubusercontent.com/5009316/235032267-1e317101-ba9b-4256-9101-5fe7a40525a1.mov

### CODE REVIEW

File: xcode/Subconscious/Shared/Components/Common/Profile/FollowUserFormSheet.swift [hunk 1]
```diff
@@ -147,15 +114,174 @@ struct FollowUserView: View {
...
+    
+    case presentQRCodeScanner(_ isPresented: Bool)
+    case qrCodeScanned(scannedContent: String)
+    case qrCodeScanError(error: String)
+}
+
+typealias FollowNewUserFormSheetEnvironment = Void
+
+// MARK: Model
+struct FollowNewUserFormSheetModel: ModelProtocol {
```
Comment (bfollington) [inline xcode/Subconscious/Shared/Components/Common/Profile/FollowUserFormSheet.swift:142]: This name is a little verbose, open to renaming this subdomain. 

It's the sheet containing the form for adding a user manually by entering a DID and petname.

File: xcode/Subconscious/Shared/Components/Common/Profile/FollowUserSheet.swift [hunk 2]
```diff
@@ -198,9 +202,6 @@ struct FollowUserSheet: View {
         }
         .padding(AppTheme.padding)
         .presentationDetents([.fraction(0.33)])
-        .onAppear {
-            send(.populate(user))
```
Comment (bfollington) [inline xcode/Subconscious/Shared/Components/Common/Profile/FollowUserSheet.swift:202]: This is unnecessary, it's dispatched by `UserProfileDetailAction.requestFollow`

File: xcode/Subconscious/Shared/Components/Common/Profile/UserProfileView.swift [hunk 3]
```diff
@@ -119,6 +119,18 @@ struct UserProfileView: View {
                         send(.presentMetaSheet(true))
                     }
                 )
+                if user.category == .you {
```
Comment (bfollington) [inline xcode/Subconscious/Shared/Components/Common/Profile/UserProfileView.swift:122]: Adds a button in the top right of your profile view to add a user manually.

File: xcode/Subconscious/Shared/Components/Common/Story/StoryUserView.swift [hunk 4]
```diff
@@ -7,6 +7,29 @@
 
 import SwiftUI
 
+/// Adjusts the hit mask of a view to exclude the top-right corner so we can add buttons there
+/// without having to deal with firing both tap targets at once.
```
Comment (bfollington) [inline xcode/Subconscious/Shared/Components/Common/Story/StoryUserView.swift:11]: This is inspired by how the same sort of interaction works on Twitter & Bluesky apps.

File: xcode/Subconscious/Shared/Config.swift [hunk 5]
```diff
@@ -26,7 +26,11 @@ struct Config: Equatable, Codable {
     /// Only returns test data when Config.debug is enabled
     var fallbackSimulatorQrCodeScanResult: String {
         get {
-            debug ? "did:key:z6MkmCJAZansQ3p1Qwx6wrF4c64yt2rcM8wMrH5Rh7DGb2K7" : ""
+            #if targetEnvironment(simulator)
+            "did:key:z6MkmCJAZansQ3p1Qwx6wrF4c64yt2rcM8wMrH5Rh7DGb2K7"
+            #else
+             ""
+            #endif
```
Comment (bfollington) [inline xcode/Subconscious/Shared/Config.swift:33]: This wasn't actually very useful before since `debug` was always off.

Comment (gordonbrander) [reply xcode/Subconscious/Shared/Config.swift:33]: Interesting! Makes me wonder if we should define a `private static let isSimulator: Bool` on Config and use that to derive properties like this?

**Review by gordonbrander — APPROVED** [review]

File: xcode/Subconscious/Shared/Components/Common/Story/StoryUserView.swift (hunk 4 above)
Comment (bfollington) [reply xcode/Subconscious/Shared/Components/Common/Story/StoryUserView.swift:11]: FYI I experimented a lot with nested buttons, it works but you have to hold down on the inner button for ~50-100ms to trigger it, a quick tap always fires the parent handler.

Found many threads telling me not to nest buttons in the first place, but after experimenting on Twitter they have to be doing something similar… either that or the “…” button is floating on top using a ZStack + .offset()

File: xcode/Subconscious/Shared/Components/Common/Profile/UserProfileView.swift [hunk 6]
```diff
@@ -68,7 +68,7 @@ struct UserProfileView: View {
         let columnFollowing = TabbedColumnItem(
             label: "Following",
             view: Group {
-                ForEach(state.following, id: \.user.did) { follow in
+                ForEach(state.following, id: \.id) { follow in
```
[the remark this answers is not in the document]
Comment (bfollington) [reply xcode/Subconscious/Shared/Components/Common/Profile/UserProfileView.swift:71]: TIL!

File: xcode/Subconscious/Shared/Config.swift (hunk 5 above)
Comment (bfollington) [reply xcode/Subconscious/Shared/Config.swift]: Added this!

**Pull request merged**